solidity/contracts/system/TBTCSystem.sol

```diff
@@ -142,13 +143,32 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
             _keepThreshold,
             _keepSize
         );
+        vendingMachine = _vendingMachine;
         setTbtcDepositToken(_tbtcDepositToken);
-        _initialized = true;
+        initialized = true;
         allowNewDeposits = true;
     }
 
-    /// @notice gets whether new deposits are allowed.
-    function getAllowNewDeposits() external view returns (bool) { return allowNewDeposits; }
+    /// @notice Returns whether new deposits should be allowed.
+    /// @return True if new deposits should be allowed, both by the emergency pause button
+    ///         and respected the max supply schedule.
+    function getAllowNewDeposits() external view returns (bool) {
+        if (!allowNewDeposits) { return false; }
```

Comment:
Could this just be `return allowNewDeposits && ...`? Understand if you settled on this approach due to a clarity preference though.

Reply:
This was actually because I assume it's cheaper gas-wise. I didn't confirm, but I wouldn't be surprised to see the Solidity compiler not properly unroll the short-circuiting

Reply:
~30 gas less to do it in one statement. Not worth the time to change it tbh.